### PR TITLE
Download one Acquia backup database at a time.

### PR DESCRIPTION
**Pull request by becw**
See #103.

This removes the old Acquia database downloading targets in favor of a new task that downloads new backups selectively via the Acquia Cloud API.

To test:
* Download your Acquia Cloud API credentials and move them to `~/.acquia/cloudapi.conf` on your Vagrant box
* Configure your Acquia site properties in `.the-build/build.default.properties.yml`:
   ```
   acquia:
     site: mysite
     realm: devcloud
   ```
* Run `phing acquia-get-backup` - the latest backup should be downloaded to `artifacts/backups`
* Run `phing acquia-load-backup` - the latest existing backup in `artifacts/backups` will be loaded in to the site db

### CODE REVIEW

File: src/TheBuild/Acquia/AcquiaTask.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      }
+
+      if (!file_exists($this->credentialsFile) || !is_readable($this->credentialsFile)) {
+        throw new BuildException("Acquia Cloud credentials file '{$this->credentialsFile}' is not available.");
+      }
+
+      $contents = file_get_contents($this->credentialsFile);
+      $creds = json_decode($contents, TRUE);
+
+      $this->email = $creds['email'];
```
[the remark this answers is not in the document]
Comment (becw) [reply src/TheBuild/Acquia/AcquiaTask.php]: Ah, I must have a very old `cloudapi.conf` file :)

File: tasks/acquia.xml [hunk 2]
```diff
@@ -20,34 +20,16 @@
...
-
-    <!-- Download production databases from Acquia -->
-    <target name="acquia-dbdl-prod">
-        <fail unless="acquia.accountname" />
-        <fail unless="acquia.ssh" />
-        <fail unless="acquia.backups" />
 
-        <filesync sourcedir="${acquia.accountname}.prod@${acquia.ssh}:/mnt/files/${acquia.accountname}.prod/backups/" destinationdir="${acquia.backups}" />
+    <target name="acquia-get-backup" description="Download the latest database backup from Acquia.">
+        <getAcquiaBackup dir="${acquia.backups}" realm="${acquia.realm}" site="${acquia.site}" env="${acquia.env}" credentialsFile="${acquia.cloud.conf}" propertyName="drupal.site.load_db.file" />
```
[the remark this answers is not in the document]
Comment (becw) [reply tasks/acquia.xml]: Yes, we should... I think I got lazy :)

File: src/TheBuild/Acquia/GetLatestBackupTask.php [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+      if ($file->exists()) {
+        $downloaded_backups[] = $backup;
+      }
+    }
+
+    // Pick out the newest current backup record, preferring already downloaded
+    // backups.
+    $newest_backup = FALSE;
+    if (!empty($downloaded_backups)) {
```
[the remark this answers is not in the document]
Comment (becw) [reply src/TheBuild/Acquia/GetLatestBackupTask.php]: I've just added the maxAge here, so you could now pass `-Dacquia.backup_age_hours=0` to force downloading the newest backup. That's a little clunky, but I'm ok with it... also, manually deleting the `backups.json` file is a fine and fairly discoverable solution, I think.

**Comment by becw** [comment]
@byrond I've fixed the bugs that you found and made the updates you suggested. I think we're ready to go now?

**Comment by becw** [comment]
Merging so that I can cut a new alpha tag.

File: src/TheBuild/Acquia/GetLatestBackupTask.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (byrond) [reply src/TheBuild/Acquia/GetLatestBackupTask.php]: This worked perfectly on 2.0.0-alpha3.  Thanks!

**Pull request merged**